### DISCUSSION TITLE
Generating the custom ECDSA Key Pair for image signing and encryption using openssl

### DISCUSSION THREAD

**Discussion by xianghui-renesas** [mcu-tools/mcuboot#1481] [Q&A] [answered]
Hi all,
I am able to generate custom ECDSA key pairs for image signing and encryption using imgtool.py. Is this capability proven to be functioning with openssl as well? I noticed the key generated using openssl looks different from what is generated from imgtool.py. 
For example, below command generate the key using openssl with SECP256R1 curve:
openssl ecparam -name prime256v1 -genkey -noout -out my_ecc_secp256r1_key.pem
Here is the comparison. On the right is the key generated with openssl:
![image](https://user-images.githubusercontent.com/55106381/192160780-a1fe998b-b4fd-45ef-acd3-33fd91509ac2.png)
Can you explain the difference in the format? Are both ok to be used with MCUboot?
I will perform some test as well.
thanks!

**Comment by utzig** [comment, marked as answer]
> Can you explain the difference in the format? Are both ok to be used with MCUboot?

OpenSSL encodes its keys in SEC1 format, while MCUboot uses PKCS#8. You can convert your generated key from the step above with this extra step:

`openssl pkcs8 -nocrypt -topk8 -in my_ecc_secp256r1_key.pem -out my_ecc_secp256r1_pkcs8_key.pem`

**Reply by xianghui-renesas** [reply]
Hi, @utzig , thanks for the explanation, this conversion works!
